dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js

```diff
@@ -0,0 +1,302 @@
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+(function() {
+  'use strict';
+
+  var module = angular.module('icgc.survival', ['icgc.donors.models']);
+
+  function noop () {}
+
+  function renderChart (params) {
+    var svg = params.svg;
+    var container = params.container;
+    var dataSets = params.dataSets;
+    var disabledDataSets = params.disabledDataSets;
+    var onMouseEnterDonor = params.onMouseEnterDonor || noop;
+    var onMouseLeaveDonor = params.onMouseLeaveDonor || noop;
+    var onClickDonor = params.onClickDonor || noop;
+    var palette = params.palette;
+
+    var containerBounds = container.getBoundingClientRect();
+
+    var yAxisLabel = 'Survival Rate';
+    var xAxisLabel = 'Duration ' + ' (days)';
+
+    var margin = {
+      top: 20,
+      right: 20,
+      bottom: 36,
+      left: 44
+    };
+    var outerWidth = containerBounds.width;
+    var outerHeight = outerWidth * 0.5;
+
+    var axisWidth = outerWidth - margin.left - margin.right;
+    var axisHeight = outerHeight - margin.top - margin.bottom;
+
+    var x = d3.scale.linear()
+      .range([0, axisWidth]);
+
+    var y = d3.scale.linear()
+      .range([axisHeight, 0]);
+
+    var xAxis = d3.svg.axis()
+      .scale(x)
+      .orient('bottom');
+
+    var yAxis = d3.svg.axis()
+      .scale(y)
+      .orient('left');
+
+    svg
+      .attr('width', outerWidth)
+      .attr('height', outerHeight)
+      .append('g')
+        .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+    var longestDuration = _.max(dataSets
+        .filter(function (data) {
+          return !_.includes(disabledDataSets, data);
+        })
+        .map(function (data) {
+          return data.donors.slice(-1)[0].survivalTime;
+        }));
+
+    x.domain([0, longestDuration]);
+    y.domain([0, 1]);
+
+    // draw x axis
+    svg.append('g')
+      .attr('class', 'x axis')
+      .attr('transform', 'translate(' + margin.left + ',' + (axisHeight + margin.top) + ')')
+      .call(xAxis)
+      .append('text')
+        .attr('class', 'axis-label')
+        .attr('dy', 30)
+        .attr('x', axisWidth / 2)
+        .style('text-anchor', 'end')
+        .text(xAxisLabel);
+
+    // draw y axis
+    svg.append('g')
+      .attr('class', 'y axis')
+      .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')')
```

Comment:
Its very common to create a wrapper for `translate` in a lot of the d3 code I've seen. Cuts down on a lot of boilerplate string concat'